app/models/MovementListSearchOptions.scala

```diff
@@ -47,41 +109,173 @@ case class MovementListSearchOptions(searchKey: Option[MovementSearchSelectOptio
           searchOptions.searchKey.map(field => stringBinder.unbind("searchKey", field.code)),
           searchOptions.searchValue.map(field => stringBinder.unbind("searchValue", field)),
           Some(stringBinder.unbind("sortBy", searchOptions.sortBy.code)),
-          Some(intBinder.unbind("index", searchOptions.index))
+          Some(intBinder.unbind("index", searchOptions.index)),
+          searchOptions.traderRole.map(field => stringBinder.unbind("traderRole", field.code)),
+          searchOptions.undischargedMovements.map(field => stringBinder.unbind("undischargedMovements", field.code)),
+          searchOptions.movementStatus.map(field => stringBinder.unbind("movementStatus", field.code)),
+          searchOptions.exciseProductCode.map(field => stringBinder.unbind("exciseProductCode", field)),
+          searchOptions.countryOfOrigin.map(field => stringBinder.unbind("countryOfOrigin", field)),
+          searchOptions.dateOfDispatchFrom.map(field => stringBinder.unbind("dateOfDispatchFrom", localDateToString(field))),
+          searchOptions.dateOfDispatchTo.map(field => stringBinder.unbind("dateOfDispatchTo", localDateToString(field))),
+          searchOptions.dateOfReceiptFrom.map(field => stringBinder.unbind("dateOfReceiptFrom", localDateToString(field))),
+          searchOptions.dateOfReceiptTo.map(field => stringBinder.unbind("dateOfReceiptTo", localDateToString(field)))
         ).flatten.mkString("&")
     }
 
-  def apply(searchKey: Option[String], searchValue: Option[String], sortBy: String): MovementListSearchOptions =
+  //noinspection ScalaStyle
+  def apply(
+             searchKey: Option[String],
+             searchValue: Option[String],
+             sortBy: String,
+             traderRoleOptions: Set[MovementFilterDirectionOption],
+             undischargedMovementsOptions: Set[MovementFilterUndischargedOption],
+             movementStatusOption: Option[MovementFilterStatusOption],
+             exciseProductCodeOption: Option[String],
+             countryOfOriginOption: Option[String],
+             dateOfDispatchFrom: Option[LocalDate],
+             dateOfDispatchTo: Option[LocalDate],
+             dateOfReceiptFrom: Option[LocalDate],
+             dateOfReceiptTo: Option[LocalDate]
+           ): MovementListSearchOptions = {
+
+    val traderRole: Option[MovementFilterDirectionOption] = {
+      (traderRoleOptions.contains(MovementFilterDirectionOption.GoodsIn), traderRoleOptions.contains(MovementFilterDirectionOption.GoodsOut)) match {
+        case (true, true) => Some(MovementFilterDirectionOption.All)
+        case (true, _) => Some(MovementFilterDirectionOption.GoodsIn)
+        case (_, true) => Some(MovementFilterDirectionOption.GoodsOut)
+        case _ => None
+      }
+    }
+
+    val undischargedMovements = undischargedMovementsOptions.toSeq match {
+      case Seq(MovementFilterUndischargedOption.Undischarged) => Some(MovementFilterUndischargedOption.Undischarged)
+      case _ => None
```

Comment:
Looks like this case isn't covered. 
Also, is there a reason for the `.toSeq` call on `undischargedMovementsOptions`?